systemtest/edu/washington/escience/myria/systemtest/UploadDownloadS3Test.java

```diff
@@ -0,0 +1,98 @@
+/**
+ *
+ */
+package edu.washington.escience.myria.systemtest;
+
+import static org.junit.Assert.assertEquals;
+
+import java.nio.file.Paths;
+import java.util.HashMap;
+
+import org.junit.Test;
+
+import edu.washington.escience.myria.CsvTupleWriter;
+import edu.washington.escience.myria.RelationKey;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.io.DataSink;
+import edu.washington.escience.myria.io.DataSource;
+import edu.washington.escience.myria.io.FileSource;
+import edu.washington.escience.myria.io.UriSink;
+import edu.washington.escience.myria.io.UriSource;
+import edu.washington.escience.myria.operator.DataOutput;
+import edu.washington.escience.myria.operator.DbInsert;
+import edu.washington.escience.myria.operator.DbQueryScan;
+import edu.washington.escience.myria.operator.FileScan;
+import edu.washington.escience.myria.operator.InMemoryOrderBy;
+import edu.washington.escience.myria.operator.RootOperator;
+import edu.washington.escience.myria.operator.network.CollectConsumer;
+import edu.washington.escience.myria.operator.network.CollectProducer;
+import edu.washington.escience.myria.operator.network.GenericShuffleConsumer;
+import edu.washington.escience.myria.operator.network.GenericShuffleProducer;
+import edu.washington.escience.myria.operator.network.partition.RoundRobinPartitionFunction;
+import edu.washington.escience.myria.operator.network.partition.SingleFieldHashPartitionFunction;
+import edu.washington.escience.myria.parallel.ExchangePairID;
+import edu.washington.escience.myria.util.JsonAPIUtils;
+
+/**
+ */
+public class UploadDownloadS3Test extends SystemTestBase {
+
+  @Test
+  public void s3UploadTest() throws Exception {
+
+    /* Ingest test data */
+    String filePath = Paths.get("testdata", "filescan", "simple_two_col_int_to_hash.txt").toString();
+    DataSource relationSource = new FileSource(filePath);
+    RelationKey relationKeyUpload = RelationKey.of("public", "adhoc", "upload");
+    Schema relationSchema = Schema.ofFields("x", Type.INT_TYPE, "y", Type.INT_TYPE);
+
+    JsonAPIUtils.ingestData("localhost", masterDaemonPort, ingest(relationKeyUpload, relationSchema, relationSource,
+        ' ', new RoundRobinPartitionFunction(workerIDs.length)));
+
+    /* File to upload and download */
+    String fileName = String.format("s3://myria-test/test.txt");
```

Comment:
We need to randomize this location with a timestamp or UUID to allow tests to be run concurrently.